packages/edit-site/src/components/page-switcher/index.js

```diff
@@ -65,21 +66,29 @@ export default function PageSwitcher( {
 		},
 		[ showOnFront ]
 	);
+
 	const onPageSelect = ( newPath ) => {
 		const { value: path, context } = [ ...pages, ...categories ].find(
 			( choice ) => choice.value === newPath
 		);
 		onActivePageChange( { path, context } );
 	};
+	const onPostSelect = ( post ) =>
+		onActivePageChange( {
```

Comment:
I found it confusing to have `settings.page` also track the data for a selected post. I wonder if something more like `settings.activeContentProvider` would be better? Maybe not that exact thing, but the goal is to set the "entity" which provides "content" for the various blocks which rely on it (post title, post content, etc)

Reply:
It still refers to the "page" under which that post loads. The page here is not the post type page; it's a front-end URL.

Reply:
> The page here is not the post type page; it's a front-end URL.

That makes more sense. I'm so used to thinking of page in the context of the CPT.